### ISSUE TITLE
[feature] Single Checkbox or Switch with value instead boolean.

### ISSUE THREAD

**Issue by igakim** [lookfirst/mui-rff#275]
Hi,
i'd like to ask why we can't set other value instead boolean when renders single checkbox or switch?

i mean:

> If you have a single checkbox, it is rendered without the label (if no label is defined) and the value is boolean.

I faced with use case when i need render groups of checkboxes separately, but with the same name.

```
<>
  <div>
    <h2>Permission group name 1</h2>
    <Checkboxes name="permissions" data={[{ label: 'Label', value: 1 }]} />
  </div>
  <div>
    <h2>Permission group name 2</h2>
    <Checkboxes name="permissions" data={[{ label: 'Label 2', value: 2 }, { label: 'Label 3', value: 3 }]} />
  </div>
</>
```

it works greatly if in each group we have more than 1 checkbox. But if in some group we have single checkbox, it set boolean value instead array of values.

i'd cloned your `Checboxes` component, and remove `single` checking and it works.

I think such case will be met quite often. 

[codesandbox](https://codesandbox.io/s/nifty-frog-7z2py?file=/src/App.js)

**Comment by thijmenverkerk** [comment]
Is there any progress with this issue? We are running into the exact same problem.

**Comment by lookfirst** [comment]
This has been fixed and released in 3.0.6
